packages/clients/src/internal/interceptors/types.ts

```diff
@@ -0,0 +1,17 @@
+export interface RequestInterceptor {
+  (request: Readonly<Request>): Request | Promise<Request>
+}
+
+export interface ResponseInterceptor {
+  (response: Readonly<Response>): Response | Promise<Response>
+}
+
+export interface ResponseErrorInterceptor {
+  (request: Request, error: unknown): unknown | Promise<unknown>
```

Comment:
related to it's will give this for this one.
```suggestion
  ({ request, error }: { request: Request, error: unknown }): unknown | Promise<unknown>
```

and inside our interceptors

```javascript
 interceptors: [
          {
            request: req => {
              return req
            },
            response: res => {

              return res
            },
            responseError: (req, error)
          },
        ],
```
this will allow autocompletion on the object and avoid mistake:

```javascript
 interceptors: [
          {
            request: ({ request }) => {
              return request
            },
            response: ({ response })  => {

              return response
            },
            responseError: ({ request, error }) => {
            
             return response
           } 
          },
        ],
```